uberfire-runtime-plugins/uberfire-runtime-plugins-client/src/main/java/org/uberfire/ext/plugin/client/info/PluginsInfo.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.uberfire.ext.plugin.client.info;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.jboss.errai.ioc.client.container.IOC;
+import org.jboss.errai.ioc.client.container.IOCBeanDef;
+import org.uberfire.client.editor.JSEditorActivity;
+import org.uberfire.client.mvp.ActivityBeansInfo;
+import org.uberfire.client.perspective.JSWorkbenchPerspectiveActivity;
+import org.uberfire.client.screen.JSWorkbenchScreenActivity;
+import org.uberfire.client.splash.JSSplashScreenActivity;
+import org.uberfire.client.workbench.type.ClientResourceType;
+import org.uberfire.client.workbench.type.ClientTypeRegistry;
+import org.uberfire.ext.plugin.client.type.DynamicMenuResourceType;
+import org.uberfire.ext.plugin.client.type.EditorPluginResourceType;
+import org.uberfire.ext.plugin.client.type.PerspectiveLayoutPluginResourceType;
+import org.uberfire.ext.plugin.client.type.ScreenPluginResourceType;
+import org.uberfire.ext.plugin.client.type.SplashPluginResourceType;
+import org.uberfire.ext.plugin.model.Activity;
+import org.uberfire.ext.plugin.model.Plugin;
+import org.uberfire.ext.plugin.model.PluginType;
+
+@ApplicationScoped
+public class PluginsInfo {
+
+    private EditorPluginResourceType editorPluginResourceType;
+
+    private PerspectiveLayoutPluginResourceType perspectiveLayoutPluginResourceType;
+
+    private ScreenPluginResourceType screenPluginResourceType;
+
+    private SplashPluginResourceType splashPluginResourceType;
+
+    private DynamicMenuResourceType dynamicMenuResourceType;
+
+    private ActivityBeansInfo activityBeansInfo;
+
+    private ClientTypeRegistry clientTypeRegistry;
+
+    protected PluginsInfo() {
```

Comment:
I'm not 100% sure, but I think tomcat requires public constructors. Please change it.